dropwizard-jersey/src/main/java/io/dropwizard/jersey/validation/ValidatingEnumParamConverterProvider.java

```diff
@@ -0,0 +1,150 @@
+package io.dropwizard.jersey.validation;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Strings;
+import com.google.common.collect.Sets;
+import io.dropwizard.jersey.DropwizardResourceConfig;
+import io.dropwizard.jersey.errors.ErrorMessage;
+import org.glassfish.jersey.server.model.Resource;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.HEAD;
+import javax.ws.rs.OPTIONS;
+import javax.ws.rs.BeanParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ParamConverter;
+import javax.ws.rs.ext.ParamConverterProvider;
+import javax.ws.rs.ext.Provider;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.EnumSet;
+import java.util.Set;
+
+/**
+ * Provides converters to jersey for enums used as resource parameters.
+ *
+ * <p>By default jersey will return a 404 if a resource parameter of an enum type cannot be converted. This class
+ * provides converters for all enum types used as resource parameters that provide better error handling. If an
+ * invalid value is provided for the parameter a {@code 400 Bad Request} is returned and the error message will
+ * include the parameter name and a list of valid values.</p>
+ */
+@Provider
+public class ValidatingEnumParamConverterProvider<C extends Enum<C>> implements ParamConverterProvider {
```

Comment:
😮 this is much better. Facepalm for the scanning work when all you need is

```java
 if (!rawType.isEnum()) {
    return null;
}
```

I will update this PR with this - should I include 

```java
 final String text = CharMatcher.WHITESPACE
                    .removeFrom(jp.getText())
                    .replace('-', '_')
                    .replace('.', '_');
```

from [`FuzzyEnumModule`](https://github.com/dropwizard/dropwizard/blob/9e01f7ec99e47f24f26698dbde7379e51b4e3aad/dropwizard-jackson/src/main/java/io/dropwizard/jackson/FuzzyEnumModule.java) to make it behave exactly the same way?